pkg/discovery/discovery_test.go

```diff
@@ -156,83 +136,70 @@ var _ = Describe("DiscoveryNG", func() {
 				})
 				go s.Start()
 				s.WaitUntilReady(timeout)
-				pb.PublishExternalEvent(playerOneIsReady, ClientIncomingEventsTopic)
-				pb.PublishExternalEvent(playerOneIsReady, ClientIncomingEventsTopic)
+				for i := 0; i < playerCount; i++ {
+					// ToDo: Isn't it an error if the same player sends its ReadyMessage multiple times?
+					//    Or rather, should we in such a case really go to PlayersReady?
+					pb.PublishExternalEvent(playerOneIsReady, ClientIncomingEventsTopic)
+				}
 				WaitDoneOrTimeout(done)
 			})
 		})
 		Context("an event from a foreign cluster is received", func() {
 			It("doesn't create a network for it", func() {
 				playersReady := GenerateEvents(PlayersReady, "0")[0]
-				playerOneIsReady := GenerateEvents(PlayerReady, "0")[0]
-				playerTwoIsReady := GenerateEvents(PlayerReady, "0")[0]
+
 				foreignFrontendAddress := "192.168.0.2"
-				player1 := proto.Player{
-					Ip: frontendAddress,
-					Id: 0,
-				}
-				player2 := proto.Player{
-					Ip: foreignFrontendAddress,
-					Id: 1,
-				}
-				playerOneIsReady.Players[0] = &player1
-				playerTwoIsReady.Players[0] = &player2
+				allPlayers, allPlayerReadyEvents := createPlayersAndPlayerReadyEvents(playerCount, foreignFrontendAddress)
+
+				allPlayers[0].Ip = frontendAddress
 
 				assertExternalEventBody(playersReady, ClientOutgoingEventsTopic, g, done, func(event *proto.Event) {
-					// Only a single port was used, 2 are still free.
-					Expect(len(n.FreePorts)).To(Equal(2))
+					// Only a single of the 6 ports was used, 5 are still free.
+					// ToDo: These Numbers are still hardcoded, adding more ports above breaks the test!
+					Expect(len(n.FreePorts)).To(Equal(6 - 1))
```

Comment:
Please initialize the free ports using

```
n = &FakeNetworker{
	FreePorts: make([]int32, playerCount),
}
for i := range n.FreePorts {
	n.FreePorts[i] = int32(30000 + i)
}
```

Then you can use `playerCount` instead of `6` here.

Reply:
✅  Done
Good Idea!